packages/server/src/util/test-utils/seed.ts

```diff
@@ -183,25 +185,36 @@ async function generateOrg(admin: User, numUsers: number, numRosterEntries: numb
 
   // Insert observations per roster entry
   let observations: Observation[] = [];
-  for (let i = 0; i < numRosterEntries; i++) {
-    observations = observations.concat(
-      observationTestData(rosterEntries[i].edipi,
-        rosterEntries[i].unit.name,
-        reportSchemas[0],
-        moment('2020-01-01T08:00:00Z'),
-        moment('2020-01-07T08:00:00Z')),
-    );
-
-    // these observations are for the single-muster config
-    // hence the same timestamp for start and end
-    observations = observations.concat(
-      observationTestData(rosterEntries[i].edipi,
-        rosterEntries[i].unit.name,
-        reportSchemas[1],
-        moment('2020-01-02T10:00:00Z'),
-        moment('2020-01-02T10:00:00Z')),
-    );
-  }
+  const rosterEntriesByUnit = _.groupBy(rosterEntries, entry => entry.unit.name);
```

Comment:
When you take a look at the `generateOrg()` function, you will see a pattern, calls to methods to generate data and then calling `save()`. However this replacement/addition does not follow the pattern. I think in either case this code would benefit from being extracted out into a function and depending on number of parameters, function name, clarity, it might be worthwhile adding a JSDoc, but that is to be seen once the code is extracted out.

Reply:
I think I misread the comment the first time...All that is happening here is that we are re-organizing the rosterEntries by their unit which is used to fill out values in a more sane manner for the observations. I do agree there is more refactoring to be done here though in general in seed.ts. I'll add this to that ticket.

Reply:
I think I misread the comment the first time...All that is happening here is that we are re-organizing the rosterEntries by their unit which is used to fill out values in a more sane manner for the observations. I do agree there is more refactoring to be done here though in general in seed.ts. I'll add this to that ticket.